django_mongodb/query_utils.py

```diff
@@ -27,25 +25,19 @@ def process_lhs(node, compiler, connection):
 def process_rhs(node, compiler, connection):
     rhs = node.rhs
     if hasattr(rhs, "as_mql"):
-        return rhs.as_mql(compiler, connection)
-    _, value = node.process_rhs(compiler, connection)
-    lookup_name = node.lookup_name
-    # Undo Lookup.get_db_prep_lookup() putting params in a list.
-    if lookup_name not in ("in", "range"):
-        value = value[0]
-    # Remove percent signs added by PatternLookup.process_rhs() for LIKE
-    # queries.
-    if lookup_name in ("startswith", "istartswith"):
-        value = value[:-1]
-    elif lookup_name in ("endswith", "iendswith"):
-        value = value[1:]
-    elif lookup_name in ("contains", "icontains"):
-        value = value[1:-1]
-
+        value = rhs.as_mql(compiler, connection)
+    else:
```

Comment:
I am wondering if we can re-write this as:
```python
if hasattr(rhs, 'as_mql'):
    ...
elif hasattr(node,"prep_lookup_value_mongo"):
   ...
else:
   # this block
   ...
```

Reply:
As it is now, `prep_lookup_value_mongo()` must be called after both the `as_mql` block (which corresponds to "Escape regex special characters if self.rhs/value is a column") and the `process_rhs` block (which corresponds to "Remove percent signs...".

I'm not very happy with this code. I want to try to eliminate `prep_lookup_value()` in the future. But we may also need more logic in this area as we complete more of the test suite, so I don't want to worry about getting it perfect right now.